plugin/src/main/java/io/fabric8/maven/plugin/SetupMojo.java

```diff
@@ -0,0 +1,268 @@
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package io.fabric8.maven.plugin;
+
+import io.fabric8.maven.core.util.MavenUtil;
+import io.fabric8.utils.DomHelper;
+import io.fabric8.utils.Strings;
+import io.fabric8.utils.XmlUtils;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProjectHelper;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.w3c.dom.Text;
+
+import java.io.File;
+import java.util.Objects;
+
+import static io.fabric8.utils.DomHelper.firstChild;
+import static org.eclipse.jgit.lib.ObjectChecker.parent;
+
+/**
+ * Ensures that the fabric8 maven plugin is defined and setup correctly in the projects pom.xml
+ */
+@Mojo(name = "setup", requiresProject = true)
+public class SetupMojo extends AbstractFabric8Mojo {
+
+    public static final String PLUGIN_GROUPID = "io.fabric8";
+    public static final String PLUGIN_ARTIFACTID = "fabric8-maven-plugin";
+    public static final String PLUGIN_DEFAULT_VERSION = "3.1.18";
+    public static final String FABRIC8_MAVEN_PLUGIN_VERSION_PROPERTY = "fabric8.maven.plugin.version";
+
+    /**
+     * Controls whether a backup pom should be created (default is true).
+     */
+    @Parameter(property = "generateBackupPoms")
+    private Boolean generateBackupPoms;
+
+    @Component
+    private MavenProjectHelper projectHelper;
+
+    @Override
+    public void executeInternal() throws MojoExecutionException, MojoFailureException {
+        File pom = project.getFile();
+        boolean updated = false;
+        Document doc;
+        try {
+            doc = XmlUtils.parseDoc(pom);
+        } catch (Exception e) {
+            getLog().error("Failed to parse pom " + pom + ". " + e, e);
+            throw new MojoExecutionException(e.getMessage(), e);
+        }
+
+        if (createOrUpdateFabric8MavenPlugin(doc)) {
+            updated = true;
+        }
+        if (updated) {
+            getLog().info("Updating the pom " + pom);
+            try {
+                DomHelper.save(doc, pom);
+            } catch (Exception e) {
+                getLog().error("Failed to update pom " + pom + ". " + e, e);
+                throw new MojoExecutionException(e.getMessage(), e);
+            }
+        }
+    }
+
+    private boolean createOrUpdateFabric8MavenPlugin(Document doc) {
+        boolean updated = false;
+        String latestVersion = MavenUtil.getVersion(PLUGIN_GROUPID, PLUGIN_ARTIFACTID, PLUGIN_DEFAULT_VERSION);
+
+        // lets check that there's a property defined
+        String currentVersion = project.getProperties().getProperty(FABRIC8_MAVEN_PLUGIN_VERSION_PROPERTY);
```

Comment:
but why not avoiding the property at all and update the `<version>...</version>` directly ?

tbh I thought about `fabric8:setup` as a one shot task for quick ramp up, but not as round-trip tool like forge. There are tons of things which could go wrong when updating or manipulating an existing setup. I.e. adding a new execution might be useful but could also be not wanted. And I don't think we should shield the user from touching the pom.xml because its not possible anyway and could raise false hope (tbh, I also dont think it should be the scope of this plugin).

A good update documentation is imo enough (as the user is already sold in using f-m-p when doing an update).

tl;dr - imo we shouldn't strive for supporting plugin updates but for one-shot setups only (with a no-op when there is already a f-m-p configured). But I'm open for discussions here.